CSharpFunctionalExtensions/Result/Extensions/Try.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace CSharpFunctionalExtensions
+{
+    public static partial class ResultExtensions
+    {
+        private static readonly Func<Exception, string> DefaultTryErrorHandler = exc => exc.Message;
+        
+        public static IResult<T, E> Try<T, E>(this IResult<T, E> result, Func<T> func, Func<Exception, E> errorHandler)
```

Comment:
As I see `result` is used only to construct another results (and it's state is not used).
I'm not sure it will look elegant in a caller code.

Same concern applies to all `Try` overloads.